File: frontend/src/services/competencyService.js
import api from "./api";

export const getCompetencies = async () => {
    const response = await api.get("/competencies");
    return response.data;
};

export const getCompetencyById = async (id) => {
    const response = await api.get(
        `/competencies/${id}`
    );
    return response.data;
};

export const getCompetenciesByEmployee = async (employeeId) => {
    const response = await api.get(
        `/competencies/employee/${employeeId}`
    );
    return response.data;
};

export const createCompetency = async (data) => {
    const response = await api.post(
        "/competencies",
        data
    );
    return response.data;
};

export const updateCompetency = async (id, data) => {
    const response = await api.put(
        `/competencies/${id}`,
        data
    );
    return response.data;
};

export const deleteCompetency = async (id) => {
    await api.delete(`/competencies/${id}`);
};

File: SkillSphere-Learning-Workspace/src/main/java/com/infosis/nexus/competency/CompetencyService.java
package com.infosis.nexus.competency;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CompetencyService {

    private final CompetencyRepository competencyRepository;

    public CompetencyService(CompetencyRepository competencyRepository) {
        this.competencyRepository = competencyRepository;
    }

    public List<Competency> getAllCompetencies() {
        return competencyRepository.findAll();
    }

    public Competency getCompetencyById(Long id) {
        return competencyRepository.findById(id)
                .orElseThrow(() ->
                        new RuntimeException("Competency not found with id: " + id));
    }

    public List<Competency> getCompetenciesByEmployee(Long employeeId) {
        return competencyRepository.findByEmployeeId(employeeId);
    }

    public Competency createCompetency(Competency competency) {
        return competencyRepository.save(competency);
    }

    public Competency updateCompetency(Long id, Competency updatedCompetency) {

        Competency existing = getCompetencyById(id);

        existing.setEmployeeId(updatedCompetency.getEmployeeId());
        existing.setCompetencyName(updatedCompetency.getCompetencyName());
        existing.setRequiredLevel(updatedCompetency.getRequiredLevel());
        existing.setCurrentLevel(updatedCompetency.getCurrentLevel());

        return competencyRepository.save(existing);
    }

    public void deleteCompetency(Long id) {
        if (!competencyRepository.existsById(id)) {
            throw new RuntimeException(
                    "Competency not found with id: " + id);
        }

        competencyRepository.deleteById(id);
    }
}